netsim/roles/router.py

```diff
@@ -0,0 +1,57 @@
+'''
+Create detailed node-level data structures from topology
+
+* Discover desired imagex (boxes)
+* Add default module list to nodes without specific modules
+* Set loopback and management interface data
+'''
+import typing
+
+from box import Box, BoxList
+
+from ..utils import log
+from ..augment import devices,addressing,links
+
+from . import select_nodes_by_role
+
+'''
+Set addresses of the main loopback interface
+'''
+def loopback_interface(n: Box, pools: Box, topology: Box) -> None:
+  n.loopback.type = 'loopback'
+  n.loopback.neighbors = []
+  n.loopback.virtual_interface = True
+  n.loopback.ifindex = 0
+  n.loopback.ifname = devices.get_loopback_name(n,topology) or 'Loopback'
+
+  pool = n.get('loopback.pool','loopback')
+  prefix_list = addressing.get(pools,[ pool ],n.id)
+
+  for af in prefix_list:
+    if prefix_list[af] is True:
+      log.error(
+        f"Address pool {pool} cannot contain unnumbered/LLA addresses",
+        category=log.IncorrectType,
+        module='nodes')
+    elif not n.loopback[af] and not (prefix_list[af] is False):
+      if af == 'ipv6':
+        if prefix_list[af].prefixlen == 128:
+          n.loopback[af] = str(prefix_list[af])
+        else:
+          n.loopback[af] = addressing.get_nth_ip_from_prefix(prefix_list[af],1)
+      else:
+        n.loopback[af] = str(prefix_list[af])
+      n.af[af] = True
+
+  for af in log.AF_LIST:
```

Comment:
Not specific to this PR, but ```log``` is a bit of a weird place to put the definition for ```AF_LIST```
```addressing``` would make more sense to me